src/main/sampling/Sample.cpp

```diff
@@ -182,6 +182,122 @@ namespace lsp
             return true;
         }
 
+        bool Sample::stretch(size_t channels, int stretch_samples, size_t chunk_size)
+        {
+            if (channels <= 0)
+                return false;
+
+            size_t max_length = nMaxLength;
+            size_t length = nLength;
+
+            // Allocate new data
+            max_length      = align_size(max_length, DEFAULT_ALIGN);    // Make multiple of 4
+            size_t to_copy      = (nMaxLength > max_length) ? max_length : nMaxLength;
```

Comment:
We can use `lsp_min` or `lsp_max` from `lsp-plug.in/common/types.h` here to simplify understanding of the ternary operator.